fbw-common/src/wasm/systems/systems/src/surveillance/egpws/runtime.rs

```diff
@@ -0,0 +1,1286 @@
+                || self.mode_2_pull_up_active)
+                && !discrete_inputs.gpws_inhibit;
+            self.alert_lamp_activated = (self.mode_1_sinkrate_lamp_active
+                || self.mode_2_pull_up_preface_active
+                || self.mode_2_terrain_active
+                || self.mode_3_lamp_active
+                || self.mode_5_glideslope_lamp_active
+                || self.mode_4_lamp_active)
+                && !discrete_inputs.gpws_inhibit;
+        } else {
+            self.warning_lamp_activated = (self.mode_1_sinkrate_lamp_active
+                || self.mode_1_pull_up_active
+                || self.mode_2_pull_up_active
+                || self.mode_2_pull_up_preface_active
+                || self.mode_2_terrain_active
+                || self.mode_3_lamp_active
+                || self.mode_4_lamp_active)
+                && !discrete_inputs.gpws_inhibit;
+            self.alert_lamp_activated =
+                self.mode_5_glideslope_lamp_active && !discrete_inputs.gpws_inhibit;
+        }
+    }
+
+    fn compute_aural_output(
+        &mut self,
+        context: &UpdateContext,
+        discrete_inputs: &TerrainAwarenessWarningSystemDiscreteInputs,
+    ) {
+        let prev_aural_output = self.aural_output;
+
+        let basic_gpws_inhibit = discrete_inputs.gpws_inhibit || discrete_inputs.audio_inhibit;
+
+        if self.mode_1_pull_up_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::PullUp;
+        } else if self.mode_2_pull_up_preface_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::Terrain;
+        } else if self.mode_2_pull_up_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::PullUp;
+        } else if self.mode_2_terrain_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::Terrain;
+        } else if self.mode_4_too_low_terrain_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::TooLowTerrain;
+        } else if self.mode_4_too_low_gear_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::TooLowGear;
+        } else if self.mode_4_too_low_flaps_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::TooLowFlaps;
+        } else if self.mode_1_sinkrate_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::SinkRate;
+        } else if self.mode_3_dont_sink_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::DontSink;
+        } else if self.mode_5_glideslope_soft_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::GlideslopeSoft;
+        } else if self.mode_5_glideslope_hard_voice_active && !basic_gpws_inhibit {
+            self.aural_output = AuralWarning::GlideslopeHard;
+        } else {
+            self.aural_output = AuralWarning::None;
+        }
```

Comment:
What about doing an initial if statement which checks if `basic_gpws_inhibit` is set and if so just returns `AuralWarning::None`? Then you don't have to check it in each following `else if` statement.
Also do `self.aural_output = if ...`.